rfcs/0098-community-team.md

```diff
@@ -0,0 +1,190 @@
+
+The community team will hold its members accountable to these ideals.
+
+## Code of Conduct and Statement of Values
+[values]: #values
+
+Following the adoption of this RFC, the code of conduct and statement of values will be displayed on a dedicated page on the NixOS website. They will read as follows, unless amended by a future RFC.
+
+The Nix community recognizes that its members come from many walks of life, and that it is natural for us to disagree on some things. Nonetheless, we are all committed to certain values, to allow us to work together and build awesome technologies.
+
+Community members are expected to follow these values in their interactions in all official Nix social spaces, including websites, chat rooms, bug threads, and conferences. We have a community team who will set and enforce social norms, to make that easier for everyone. faCommunity members are expected to follow the community team's requests.
+
+If you have opinions on the community team's work, get involved! It's a rotating team, and we want to develop the leadership potential in all community members.
+
+Our values are as follows.
+
+**Collaboration** — We all want to make the best technologies we can. We work together on our common goals, with open discussion, and without petty infighting or focusing on blame. We are all, always, teaching each other to be better.
+
+**Communication** — When we encounter problems, we focus on solving them, not on fighting each other. When we disagree on technical subjects we follow appropriate processes to escalate them to technical leadership. We try to avoid and de-escalate personal conflicts, but when they are unavoidable we seek mediation.
+
+**Empathy** — We try to see perspectives other than our own, to treat everyone with kindness and empathy, to move past mistakes. When we criticise each other we try to do so graciously. We respect each other's different backgrounds and viewpoints, and we respect the effort everyone puts into the community.
+
+**Humility** — We acknowledge that everybody has something to learn, that nobody is infallible.
+
+**Inclusion** — We work to include everybody, especially people from marginalized backgrounds. We strive to be fair and equitable, and to respect the ways people are different. We reject bigotry in all its forms.
+
+**Accommodation** — When community members face barriers to participation, we accommodate their needs to the best of our ability. This includes, but is not limited to, needs related to disability either physical or mental, to medical issues, to religious observations, to parenting, and to sober lifestyles.
```

Comment:
Should this point acknowledge that an important issue of accomodation is how much preexisting knowledge of «generic OS use» is expected or effectively expected, and how this expectation is communicated? Maybe with an immediate out-of-scoping as an issue that has too much technical weight.

Reply:
To be sure I understand: Are you saying that there should be something about not accommodating people who don't have a minimum level of background knowledge? Or are you saying the opposite, that we should be explicit about *not* requiring background knowledge?

Reply:
On the object level, I think we currently have need for some background knowledge, and some of the community members are working on removing accidental prerequisites, so to say.

On the value level, I think clearly admitting in advance that we will fail to accomodate people lacking too much compared to the current bar is a necessity if only out of honesty. I guess praising here the efforts of those trying to lower the bar would make it clear that explaining basics to someone clearly but not critically underskilled is not flooding the channel with noise but a valuable effort…

Reply:
I think there's a balance here. To whatever extent we have people who are wandering in who are actual children, trying to take their first steps in software development, I absolutely think that any community member who wants to use their own time to help with that should be encouraged to. We have to put some effort into teaching people, or our community will be self-limiting and won't endure.

Of course, on the other end of the scale, if somebody's actual paid job is to develop software, and they're taking advantage of people's good will to do the work for them, I don't think that's a good use of community resources. Flooding the channel is misusing a shared resource, and at some point the community team needs to intervene and stop that.

I don't think we need to say anything concrete about exactly where we draw that line. Our values are things we aspire to, not a statement of things we will or won't do.

Reply:
Then again, I would expect children learning to program to have more background knowledge than someone with a non-STEM education and non-STEM job who distro-hopped as a hobby and took NixOS as a challenge. The balance of considerations in helping a Windows programmer move to Linux can also be complicated.

I think it is fine to say that the level of accomodation in the context of need of preexisting knowledge is out of scope. I think talking about all the reasons to need some accomodation but never acknowledging the existence of the one thing that comes up most is at least a bit strange.

Reply:
> I think it is fine to say that the level of accomodation in the context of need of preexisting knowledge is out of scope.

Who is the audience here? Do we expect this statement of values to deter some hypothetical question-asker? Do we want to embolden experienced Nix users to say, "Sorry, it's clear you don't know enough. Please go back to the basics"?

Reply:
First of all there is a hope to encourage someone to write honest prerequisite knowledge section for the documentation…

But at some point asking a newcomer «do you know X, Y, and Z at all» and recommending to learn this in a non-Nix-specific way is better than the realistic (and perfectly good faith) alternatives.

Reply:
in any case, i don't think something like this belongs in our statement of values. if the nixos manual should include recommended reading material or background knowledge, that is a separate thing.

Reply:
I was about to try to simplify this bullet point, but I'm not clear on what this adds beyond the values listed before it. Remove?

Reply:
We're not removing "accommodation", it's core. It implies, for example, that we wouldn't do something important on a video call without finding out whether video is possible for everyone (blind or deaf community members might need captions, or people might not be able to attend at the time we choose). That's what it adds.

Reply:
Does this effectively imply that holding NixCon as an official community event is wrong?

Reply:
That's a great question, and I really appreciate you bringing it up, I was hoping someone would.

Not at all! NixCon is important, and we should absolutely have it. The "accommodation" value does imply that we would do what we can to provide accessibility at the con, and to not schedule it against any important religious holidays or anything like that. Of course, there are budgetary constraints, and logistical ones, which the Foundation - not the community team - would have the final call on. I've seen many technical conferences successfully navigate accessibility concerns and I am confident the Nix community can do it too.

Reply:
While the value is definitely useful in a border sense, I wonder if it really adds guidlines in the potentially conflictong areas that a Community (Moderation) Team would be charhed to bring jurisdiction to.

I think it's critically important aspects, such as the ones you mention are better safeguarded and protocolized in an ongoing evoluationary manner out of current practices.

Reply:
It's not the "community moderation team", it's the "community team". Its duties include moderation, leadership development, and generally fostering the growth of the community. That's explicitly a larger role than just moderation, and by design. The leadership development element is crucial to making this work, as I've explained elsewhere in the thread - it's the only way this can scale.

Accessibility is a key area which is often contentious for communities, and if we fail to address it in the statement of values now, it will be contentious every time it comes up, forever. By explicitly bringing it up now, we get to move past those conversations.

Reply:
I support the "Community Moderation Team" verbiage by @blaggacao.

While I'd love to address the additional issues in the community about leadership development and fostering the growth of the community, those are out of scope for this RFC. I specifically care about this point because it expands the role beyond the scope of the discussion and original intent. In fact, we should ensure that this team does not de-facto become leaders of the community in realms other than communication moderation. I fully agree with and refer to the stated non-goal: "Create a hierarchical power structure".

Supporting commentary: from the summary:
```
Establish an official community team to make moderation decisions, model social norms, and mediate interpersonal problems.
```
Moderation, Model, Mediate. Much of that verbiage and in other sections refers to moderation. I interpret this RFC to discuss a team to mediate only non-technical issues that come up in the various forum/chat/spaces in the community. 

(I'm not sure if what was meant by "leadership development" is both non-technical and technical leadership or only one of them. My overall interpretation of this RFC that it addresses non-technical moderation and leadership abiding by the values and enacted by the rotating set of people on the team.)

Reply:
Actually from the current formulation you cannot even know if this bullet point is work mandate or acceptance mandate. For medium meetings you use as an example, verifying everyone who is expected to be able to participate can actually participate is one thing. For large things like NixCon there is quite a difference between «make accept trade-offs that come when someone offers to improve accessibility» and «do not expand the scope of work if you don't correspondingly excpand the scope of accesibility work».

Reply:
This is in response to @tomberek.

> I support the "Community Moderation Team" verbiage by @blaggacao.
> 
> While I'd love to address the additional issues in the community about leadership development and fostering the growth of the community, those are out of scope for this RFC. I specifically care about this point because it expands the role beyond the scope of the discussion and original intent. In fact, we should ensure that this team does not de-facto become leaders of the community in realms other than communication moderation. I fully agree with and refer to the stated non-goal: "Create a hierarchical power structure".

It's explicitly in scope for the RFC. It's bullet point 3 in goals, "Foster newcomers and develop their leadership potential."

I can imagine an RFC that attempted to deal only with the narrower concept of moderation, but it is not this RFC. Everything about this RFC relies on the feedback loop, with community growth and trust feeding into each other, that the focus on leadership development creates.

> 
> Supporting commentary: from the summary:
> 
> ```
> Establish an official community team to make moderation decisions, model social norms, and mediate interpersonal problems.
> ```
> 
> Moderation, Model, Mediate. Much of that verbiage and in other sections refers to moderation. I interpret this RFC to discuss a team to mediate only non-technical issues that come up in the various forum/chat/spaces in the community.
> 
> (I'm not sure if what was meant by "leadership development" is both non-technical and technical leadership or only one of them. My overall interpretation of this RFC that it addresses non-technical moderation and leadership abiding by the values and enacted by the rotating set of people on the team.)

The first listed non-goal is "Provide technical leadership". Technical leadership is out of scope for this RFC.

That is not to say that technical disputes are entirely out of scope. To the extent that such disputes become acrimonious in ways which contradict the statement of values, the community team would have the role of guiding things back on to track so that escalating to technical leadership can proceed as it should. I think technical leadership right now is mostly edolstra? Whoever it is, this RFC doesn't change that.

I hope that, even though we disagree about whether there should be a hierarchical power structure, you can appreciate my honesty in putting that topic front and center.

Reply:
Hmmm, maybe you should look up the «RFC SC rotation» discussion in this repository, I think you might make some useful observations about the community dynamics in this project specifically.

Reply:
> Hmmm, maybe you should look up the «RFC SC rotation» discussion in this repository, I think you might make some useful observations about the community dynamics in this project specifically.

@7c6f434c do you mean https://github.com/NixOS/rfcs/pull/43 ? Most of what I get from that discussion is that it was very civil and uncontroversial, which is a good thing. Was that your point?

@blaggacao It's sure been an adventure.

Reply:
> #43

Well, first of all if you want to know how technical leadership works, that's there. Second, if you want to know how process RFCs normally get handled here (get pared down to the very bare bones, then it turns out there is still a lot to carefully polish), you might also be interested in the latest example.

Reply:
Ah, that may well be a useful example, yes. Thank you.